File: WorldSimulatorCPP/Dandelion.cpp
#include "Dandelion.h"
#include "World.h"
Dandelion::Dandelion(World* w, Coordinate c) {
	strength = 0;
	initiative = 0;
	species = "Dandelion";
	sprite = 'd';
	age = 0;
	world = w;
	coord = c;
	dead = false;
	w->addOrganism(this);
}

Dandelion::Dandelion(World* w, Coordinate c, int s, int a) {
	strength = s;
	initiative = 0;
	species = "Dandelion";
	sprite = 'd';
	age = a;
	world = w;
	coord = c;
	dead = false;
	w->addOrganism(this);
}

void Dandelion::action() {
	bool grew = false;
	int growTimes = 0;
	while (growTimes < 3) {
		int growRoll = rand() % 100;
		if (growRoll <= GROW_CHANCE) {
			grew = true;
			Coordinate newbornCoord = FindVoidSpace(getCoord());
			if (newbornCoord != getCoord()) {
				multiply(this, newbornCoord);
				cout << getSpecies() << getID() << " z pola " << getCoord() << " rozprzestrzenia sie na pole " << newbornCoord << endl;
			}
			else cout << getSpecies() << getID() << " z pola " << getCoord() << " nie moze sie rozprzestrzenic - brak pustego miejsca w poblizu" << endl;
		}
		growTimes++;
	}
	if (grew)
		cout << endl;
}

void Dandelion::draw() const{
	cout << BRIGHT_YELLOW << getSprite() << RESET;
}

File: WorldSimulatorJAVA/Dandelion.java
package worldsimulator;

import java.util.Random;

public class Dandelion extends Plant {
    

    public Dandelion(World w, Coordinate c) {
    	strength = Species.DANDELION.getStrength();
        initiative = Species.DANDELION.getInitiative();
        species = Species.DANDELION.getName();
        sprite = Species.DANDELION.getSprite();
        age = 0;
        world = w;
        coord = c;
        dead = false;
        world.addOrganism(this);
    }

    public Dandelion(World w, Coordinate c, int s, int a) {
    	strength = s;
    	initiative = Species.DANDELION.getInitiative();
        species = Species.DANDELION.getName();
        sprite = Species.DANDELION.getSprite();
        age = a;
        world = w;
        coord = c;
        dead = false;
        world.addOrganism(this);
    }

    @Override
    public void action() {
    	String message;
        boolean grew = false;
        int growTimes = 0;
        Random rand = new Random();
        while (growTimes < 3) {
            int growRoll = rand.nextInt(100);
            if (growRoll <= GROW_CHANCE) {
                grew = true;
                Coordinate newbornCoord = findVoidSpace(getCoord());
                if (!newbornCoord.equals(getCoord())) {
                    multiply(this, newbornCoord);
                    message = String.format(getSpecies() + getID() + " z pola " + getCoord() + " rozprzestrzenia sie na pole " + newbornCoord);
                    world.prompt(message);
                } else {
                	message = String.format(getSpecies() + getID() + " z pola " + getCoord() + " nie moze sie rozprzestrzenic - brak pustego miejsca w poblizu");
                	world.prompt(message);
                }
            }
            growTimes++;
        }
        if (grew) {
        	message = String.format(" ");
        	world.prompt(message);
        }
    }
}
